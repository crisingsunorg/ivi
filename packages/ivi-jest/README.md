# Jest utils for ivi library
