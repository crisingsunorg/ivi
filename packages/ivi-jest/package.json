{
  "private": true,
  "name": "ivi-jest",
  "version": "0.0.1",
  "main": "dist/main/index.js",
  "module": "dist/module/index.js",
  "sideEffects": false,
  "files": [
    "dist",
    "src",
    "!dist/**/*.tsbuildinfo",
    "!src/**/__tests__",
    "README.md",
    "LICENSE"
  ],
  "scripts": {
    "prepublishOnly": "npm run clean && npm run dist",
    "clean": "rm -rf ./dist",
    "dist": "npm run dist:main && npm run dist:module",
    "dist:main": "../../node_modules/.bin/tsc -p ./tsconfig.build.main.json",
    "dist:module": "../../node_modules/.bin/tsc -p ./tsconfig.build.module.json"
  },
  "dependencies": {
    "jsdom": "^14.0.0"
  },
  "peerDependencies": {},
  "devDependencies": {},
  "description": "Jest utils for ivi library",
  "license": "MIT",
  "author": {
    "name": "Boris Kaul",
    "email": "localvoid@gmail.com",
    "url": "https://github.com/localvoid"
  },
  "keywords": [
    "ivi"
  ],
  "homepage": "https://github.com/localvoid/ivi",
  "repository": {
    "type": "git",
    "url": "https://github.com/localvoid/ivi"
  },
  "bugs": {
    "url": "https://github.com/localvoid/ivi/issues"
  },
  "github": "https://github.com/localvoid/ivi"
}
