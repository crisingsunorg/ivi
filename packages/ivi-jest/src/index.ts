export { useMockFn, useSpyOn } from "./mock";
export { useResetModules, useModule } from "./module";
export { useDOMElement, useDOMOpsCounters } from "./dom";
