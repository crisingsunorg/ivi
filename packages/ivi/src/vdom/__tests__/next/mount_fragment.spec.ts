import { useResetModules, useModule, useDOMElement, useDOMOpsCounters } from "ivi-jest";

useResetModules();
const c = useDOMElement();
const domOps = useDOMOpsCounters();
const h = useModule<typeof import("ivi-html")>("ivi-html");
const t = useModule<typeof import("ivi-test")>("ivi-test");

describe("mount fragment", () => {
  test("empty", () => {
    const { domNode } = t.render([], c());
    expect(domNode).toMatchSnapshot();
    expect(domOps()).toMatchSnapshot();
  });

  test("single child", () => {
    const { domNode } = t.render([h.h1()], c());
    expect(domNode).toMatchSnapshot();
    expect(domOps()).toMatchSnapshot();
  });

  test("two children", () => {
    const { domNode } = t.render([h.h1(), h.h2()], c());
    expect(domNode).toMatchSnapshot();
    expect(domOps()).toMatchSnapshot();
  });

  test("null between two children", () => {
    const { domNode } = t.render([h.h1(), null, h.h2()], c());
    expect(domNode).toMatchSnapshot();
    expect(domOps()).toMatchSnapshot();
  });
});
