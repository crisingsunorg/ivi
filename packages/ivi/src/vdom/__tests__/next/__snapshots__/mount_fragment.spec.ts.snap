// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount fragment empty 1`] = `null`;

exports[`mount fragment empty 2`] = `
Object {
  "appendChild": 0,
  "createElement": 0,
  "createElementNS": 0,
  "createTextNode": 0,
  "innerHTML": 0,
  "insertBefore": 0,
  "nodeValue": 0,
  "removeChild": 0,
  "replaceChild": 0,
  "textContent": 0,
}
`;

exports[`mount fragment null between two children 1`] = `<h1 />`;

exports[`mount fragment null between two children 2`] = `
Object {
  "appendChild": 0,
  "createElement": 2,
  "createElementNS": 0,
  "createTextNode": 0,
  "innerHTML": 0,
  "insertBefore": 2,
  "nodeValue": 0,
  "removeChild": 0,
  "replaceChild": 0,
  "textContent": 0,
}
`;

exports[`mount fragment single child 1`] = `<h1 />`;

exports[`mount fragment single child 2`] = `
Object {
  "appendChild": 0,
  "createElement": 1,
  "createElementNS": 0,
  "createTextNode": 0,
  "innerHTML": 0,
  "insertBefore": 1,
  "nodeValue": 0,
  "removeChild": 0,
  "replaceChild": 0,
  "textContent": 0,
}
`;

exports[`mount fragment two children 1`] = `<h1 />`;

exports[`mount fragment two children 2`] = `
Object {
  "appendChild": 0,
  "createElement": 2,
  "createElementNS": 0,
  "createTextNode": 0,
  "innerHTML": 0,
  "insertBefore": 2,
  "nodeValue": 0,
  "removeChild": 0,
  "replaceChild": 0,
  "textContent": 0,
}
`;
